cmd/kube-controller-manager/app/core.go

```diff
@@ -179,11 +198,25 @@ func startRouteController(ctx ControllerContext) (http.Handler, bool, error) {
 		klog.Warning("configure-cloud-routes is set, but cloud provider does not support routes. Will not configure cloud provider routes.")
 		return nil, false, nil
 	}
-	_, clusterCIDR, err := net.ParseCIDR(ctx.ComponentConfig.KubeCloudShared.ClusterCIDR)
-	if err != nil {
-		klog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.ComponentConfig.KubeCloudShared.ClusterCIDR, err)
+
+	clusterCIDRList := strings.Split(ctx.ComponentConfig.KubeCloudShared.ClusterCIDR, ",")
+	clusterCIDRs := make([]*net.IPNet, 0)
+	for idx, thisCIDR := range clusterCIDRList {
+		if idx > 0 && !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.IPv6DualStack) {
+			klog.Warningf("multiple cluster cidrs were supplied but dual stack is not enabled, ignoring all but first")
+			break
+		}
+		if idx > 1 {
+			klog.Warningf("dual stacks enabled but more than 2 cidrs were supplied. ignoring all but first two cidrs")
+			break
+		}
+		_, clusterCIDR, err := net.ParseCIDR(thisCIDR)
+		if err != nil {
+			klog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.ComponentConfig.KubeCloudShared.ClusterCIDR, err)
+		}
+		clusterCIDRs = append(clusterCIDRs, clusterCIDR)
```

Comment:
it should actually `die`, otherwise the cluster will boot-up with wrong set/order of cidrs and node cidr allocation will be funky, users who do this mistake (if it does not die) will have to delete all node objects, all pods, then reset controller manager.

I changed both `controller-manager` and `cloud-controller-manager` to die if one-of ClusterCIDRs is un-parse-able